configure.ac

```diff
@@ -57,30 +85,88 @@ AC_ARG_WITH([state-file-path],
         STATE_FILE_PATH="$withval"
         ;;
     esac
-  ],
-  [
-    STATE_FILE_PATH="/var/lib/logrotate.status"
   ])
 AC_DEFINE_UNQUOTED([STATEFILE], "$STATE_FILE_PATH")
 AC_SUBST(STATE_FILE_PATH)
 
 AC_ARG_WITH([default-mail-command],
   AC_HELP_STRING([--with-default-mail-command=COMMAND],
-                 [default mail command (based on config.h if not given)]),
+                 [default mail command (default: /bin/mail -s)]),
   [
     case "$withval" in
       yes|no)
         AC_MSG_ERROR([--with-default-mail-command=COMMAND requires a command to be specified])
         ;;
       *)
         DEFAULT_MAIL_COMMAND="$withval"
-        AC_DEFINE_UNQUOTED([DEFAULT_MAIL_COMMAND], "$DEFAULT_MAIL_COMMAND")
-        AC_SUBST(DEFAULT_MAIL_COMMAND)
         ;;
     esac
   ])
+AC_DEFINE_UNQUOTED([DEFAULT_MAIL_COMMAND], "$DEFAULT_MAIL_COMMAND")
+AC_SUBST(DEFAULT_MAIL_COMMAND)
+
+AC_ARG_WITH([compress-command],
+  AC_HELP_STRING([--with-compress-command=COMMAND],
+                 [compress command (default: /bin/gzip)]),
+  [
+    case "$withval" in
+      yes|no)
+        AC_MSG_ERROR([--with-compress-command=COMMAND requires a command to be specified])
+        ;;
+      *)
+        COMPRESS_COMMAND="$withval"
+        ;;
+    esac
+  ])
+AC_DEFINE_UNQUOTED([COMPRESS_COMMAND], "$COMPRESS_COMMAND")
+AC_SUBST(COMPRESS_COMMAND)
+
+AC_ARG_WITH([uncompress-command],
+  AC_HELP_STRING([--with-uncompress-command=COMMAND],
+                 [uncompress command (default: /bin/gunzip)]),
+  [
+    case "$withval" in
+      yes|no)
+        AC_MSG_ERROR([--with-uncompress-command=COMMAND requires a command to be specified])
+        ;;
+      *)
+        UNCOMPRESS_COMMAND="$withval"
+        ;;
+    esac
+  ])
+AC_DEFINE_UNQUOTED([UNCOMPRESS_COMMAND], "$UNCOMPRESS_COMMAND")
+AC_SUBST(UNCOMPRESS_COMMAND)
+
+AC_ARG_WITH([compress-extension],
+  AC_HELP_STRING([--with-compress-extension=EXTENSION],
+                 [compress extension (default: .gz)]),
+  [
+    case "$withval" in
+      yes|no)
+        AC_MSG_ERROR([--with-compress-extension=EXTENSION requires an extension to be specified])
```

Comment:
however, perhaps just test that the value begins with dot:

```
COMPRESS_EXT=".gz"
```

so that `--with-compress-extension=gz` is invalid, but `--with-compress-extension=.gz` is okay